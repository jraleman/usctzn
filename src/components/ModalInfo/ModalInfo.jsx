import React from 'react';
import styled from 'styled-components';

const ModalInfo = ({ isModalOpen }) => {
    if (!isModalOpen) {
        return null;
    }
    return (
        <ModalWrapper>
            <ModalContainer>
                <Title>About</Title>
                <InfoContent>
                    <strong>👨‍💻 Author: </strong> Jose Ramon Aleman
                    <br />
                    <strong>📝 Description: </strong> Get prepared for the citizenship interview :D
                </InfoContent>
            </ModalContainer>
        </ModalWrapper>
    );
};

const ModalWrapper = styled.div`
    position: absolute;
    left: 0;
    right: 0;
    height: 100vh;
    background:black;
    background:rgba(0,0,0,0.8);
`;

const ModalContainer = styled.div`
    text-align: center;
    color: ${({ theme }) => theme.color.modal.text};
    border: ${({ theme }) => theme.color.modal.border} solid 1px;
    background-color: ${({ theme }) => theme.color.modal.background};
    margin: 0 auto;
    max-width: 500px;
`;

const Title = styled.h2``;

const InfoContent = styled.p`
    padding: 1em;
`;

export default ModalInfo;
