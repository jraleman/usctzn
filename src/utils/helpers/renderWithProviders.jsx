import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../styles/theme';

const renderWithProviders = (Component) => {
    render(
        <ThemeProvider theme={lightTheme}>
            {Component}
        </ThemeProvider>
    )
};

export default renderWithProviders;
