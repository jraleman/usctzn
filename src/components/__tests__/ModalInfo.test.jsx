import React from 'react';
import renderWithProviders from '../../utils/helpers/renderWithProviders';
import ModalInfo from '../ModalInfo';

describe('ModalInfo', () => {
    it('renders', async () => {
        const component = renderWithProviders(<ModalInfo  />);
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });
});
