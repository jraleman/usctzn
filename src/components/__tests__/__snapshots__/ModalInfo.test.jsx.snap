// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalInfo renders 1`] = `undefined`;
